### PR TITLE
Int 26 - S narrative will create and populate the screener narrative word document the legacy

### PR DESCRIPTION
**Pull request by igor-chornobay-engagepoint**
CmsDocument endpoint with CRUD methods. 
Compression:
LZW - 
PKware - 
PLAINE - (test only)

Stub to create simplest word document from text.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java [hunk 1]
```diff
@@ -59,21 +78,25 @@ public CmsDocumentDao(@CmsSessionFactory SessionFactory sessionFactory) {
...
-      Splitter.fixedLength(4000).split(hex).forEach(segments::add);
+      final byte[] compressed = new CmsPKCompressor().compressBytes(bytes);
 
       int i = 0;
-      for (String docBlob : segments) {
+      int segmentStart = 0;
+      while (segmentStart < compressed.length) {
         final String sequence = StringUtils.leftPad(String.valueOf(++i), 4, '0');
-        blobs.add(new CmsDocumentBlobSegment(doc.getId(), sequence, docBlob));
+        final int segmentLenght = min(compressed.length - segmentStart, BLOB_SEGMENT_LENGTH);
```
Comment (ThomasOSI) [inline src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java]: should `segmentLenght` be `segmentLength`?

Comment (igor-chornobay-engagepoint) [reply src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java]: yeah. typo..

Comment (igor-chornobay-engagepoint) [reply src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java]: fixed

File: src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java [hunk 2]
```diff
@@ -87,6 +110,43 @@ public CmsDocumentDao(@CmsSessionFactory SessionFactory sessionFactory) {
...
     return blobs;
   }
+  /**
+   * No compressed document blob segments.
+   *
+   * @param doc the document record
+   * @param base64 base64 encoded bytes
+   * @return new blobs in order
+   */
+  protected List<CmsDocumentBlobSegment> compressPLAIN(final CmsDocument doc, String base64) {
```
Comment (ThomasOSI) [inline src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java]: Curious why protected is specified. Do we have sub-classes (or package classes) that need to see this function?

File: src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java [hunk 3]
```diff
@@ -178,12 +299,18 @@ protected String decompressLZW(CmsDocument doc) {
...
 
-    try (FileOutputStream fos = new FileOutputStream(src);) {
-      for (CmsDocumentBlobSegment seg : doc.getBlobSegments()) {
-        final byte[] bytes = DatatypeConverter.parseHexBinary(seg.getDocBlob().trim());
-        fos.write(bytes, 0, bytes.length);
+    try {
+      try (FileOutputStream fos = new FileOutputStream(src);){
+        for (CmsDocumentBlobSegment seg : doc.getBlobSegments()) {
+          final byte[] bytes = seg.getDocBlob();
+//          fos.write(bytes, 0, bytes.length);
```
Comment (ThomasOSI) [inline src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java]: Recommend removing code that has been commented out.

File: src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java (hunk 2 above)
Comment (igor-chornobay-engagepoint) [reply src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java]: should be private really.

File: src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java [hunk 4]
```diff
@@ -206,9 +333,132 @@ protected String decompressLZW(CmsDocument doc) {
...
+  }
+
+
+
+  public byte[] createWordBytes(String textBody) {
+
+    try (XWPFDocument doc = new XWPFDocument()) {
+
+      XWPFParagraph paragraph = doc.createParagraph();
+//      paragraph.setAlignment(ParagraphAlignment.LEFT);
```
Comment (ThomasOSI) [inline src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java]: Recommend removing commented out code.

File: src/main/java/gov/ca/cwds/data/persistence/cms/CmsDocumentBlobSegment.java [hunk 5]
```diff
@@ -51,7 +52,7 @@
   @Size(min = 1, max = 4000)
   // @ColumnTransformer(read = "blob(DOC_BLOB)", write = "x'?'")
```
Comment (ThomasOSI) [inline src/main/java/gov/ca/cwds/data/persistence/cms/CmsDocumentBlobSegment.java]: Recommend removing lines that have been commented out.

File: src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java (hunk 2 above)
Comment (igor-chornobay-engagepoint) [reply src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java]: fixed

File: src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java (hunk 4 above)
Comment (igor-chornobay-engagepoint) [reply src/main/java/gov/ca/cwds/data/cms/CmsDocumentDao.java]: fixed

File: src/main/java/gov/ca/cwds/data/persistence/cms/CmsDocumentBlobSegment.java (hunk 5 above)
Comment (igor-chornobay-engagepoint) [reply src/main/java/gov/ca/cwds/data/persistence/cms/CmsDocumentBlobSegment.java]: fixed

**Pull request merged**